libs/agno/agno/tools/whatsapp.py

```diff
@@ -0,0 +1,212 @@
+import json
+import os
+from typing import Any, Dict, List, Optional
+
+import httpx
+from dotenv import load_dotenv
+
+from agno.tools import Toolkit
+from agno.utils.log import logger
+
+# Try to load from both .env and .envrc
+load_dotenv()
+
+
+class WhatsAppTools(Toolkit):
+    """WhatsApp Business API toolkit for sending messages."""
+
+    base_url = "https://graph.facebook.com"
+
+    def __init__(
+        self,
+        access_token: Optional[str] = None,
+        phone_number_id: Optional[str] = None,
+        version: str = "v22.0",
+        recipient_waid: Optional[str] = None,
+    ):
+        """Initialize WhatsApp toolkit.
+
+        Args:
+            access_token: WhatsApp Business API access token
+            phone_number_id: WhatsApp Business Account phone number ID
+            version: API version to use
+            recipient_waid: Default recipient WhatsApp ID (optional)
+        """
+        super().__init__(name="whatsapp")
+
+        # Core credentials
+        self.access_token = access_token or os.getenv("WHATSAPP_ACCESS_TOKEN") or os.getenv("ACCESS_TOKEN")
```

Comment:
```suggestion
        self.access_token = access_token or os.getenv("WHATSAPP_ACCESS_TOKEN")
```
We might want that for other tools?